docs/parameters/validation.md

```diff
@@ -0,0 +1,136 @@
+
+### Example
+
+The example below has a deliberate typo in `params.input` (`.txt` instead of `.csv`).
+The validation function catches this for two reasons:
+
+- The filename doesn't match the expected pattern (here checking file extensions)
+- The supplied file doesn't exist
+
+The function causes Nextflow to exit immediately with an error.
+
+=== ":material-close-thick: Output"
+
+    ```
+    --8<-- "examples/validateParameters/log.txt"
+    ```
+
+=== "main.nf"
+
+    ```groovy
+    --8<-- "examples/validateParameters/pipeline/main.nf"
+    ```
+
+=== "nextflow.config"
+
+    ```groovy hl_lines="6"
+    --8<-- "examples/validateParameters/pipeline/nextflow.config"
+    ```
+
+=== "nextflow_schema.json"
+
+    ```json linenums="1" hl_lines="17 20"
+    --8<-- "examples/validateParameters/pipeline/nextflow_schema.json"
+    ```
+
+## Unrecognized parameters
+
+When parameters which are not specified in the JSON Schema are provided, the parameter validation function returns a `WARNING`.
+This is because user-specific institutional configuration profiles may make use of params that are unknown to the pipeline.
+
+The down-side of this is that warnings about typos in parameters can go unnoticed.
+
+To force the pipeline execution fail with an error instead, you can provide the `validationFailUnrecognisedParams` parameter:
+
+```bash
+nextflow run my_pipeline --validationFailUnrecognisedParams
+```
+
+or specify it in the configuration file
+
+```groovy
+params.validationFailUnrecognisedParams = true
+```
+
+=== ":material-check: Default"
+
+    === "Output"
+
+        ```
+        --8<-- "examples/validationWarnUnrecognisedParams/log.txt"
+        ```
+
+    === "nextflow.config"
+
+        ```groovy
+        --8<-- "examples/validationWarnUnrecognisedParams/pipeline/nextflow.config"
+        ```
+    === "main.nf"
+
+        ```groovy
+        --8<-- "examples/validationWarnUnrecognisedParams/pipeline/main.nf"
+        ```
+
+=== ":material-close-thick: Fail unrecognised params"
+
+    === "Output"
+
+        ```
+        --8<-- "examples/validationFailUnrecognisedParams/log.txt"
+        ```
+
+    === "nextflow.config"
+
+        ```groovy hl_lines="6"
+        --8<-- "examples/validationFailUnrecognisedParams/pipeline/nextflow.config"
+        ```
+    === "main.nf"
+
+        ```groovy
+        --8<-- "examples/validationFailUnrecognisedParams/pipeline/main.nf"
+        ```
+
+## Variable type checking
+
+By default, `validateParameters()` is strict about expecting parameters to adhere to their expected type.
+If the schema says that `params.foo` should be an `integer` and the user sets `params.foo = "12"` (a string with a number), it will fail.
+
+If this is a problem, the user can run validation in "lenient mode", whereby the JSON Schema validation tries to _cast_ parameters to their correct type.
+For example, providing an integer as a string will no longer fail validation.
```

Comment:
and does it convert the string to integer?

Reply:
it does :) It tries to cast all provided values to their corresponding type specified in the schema.

Reply:
No the function does not return anything, it's only used for validation

Reply:
Oh, maybe I misunderstood the question, I meant that the lenient mode can validate strings that should be values and the other way around. But as @nvnieuwk said, it doesn't return any value.

Reply:
I was unclear on this, so it doesn't modify the params in place then? A string stays as a string even if it validates as an int?

Reply:
Added a note to clarify this.